libcodechecker/server/client_db_access_handler.py

```diff
@@ -695,6 +696,145 @@ def getSuppressedBugs(self, run_id):
             raise shared.ttypes.RequestFailed(shared.ttypes.ErrorCode.IOERROR,
                                               msg)
 
+    @timeit
+    def getComments(self, bug_hash):
+        """
+            Return the list of comments for the given bug.
+        """
+        session = self.__session
+        try:
+            result = []
+
+            comments = session.query(Comment) \
+                .filter(Comment.bug_hash == bug_hash) \
+                .order_by(Comment.created_at.desc()) \
+                .all()
+
+            for comment in comments:
+                result.append(CommentData(
+                    comment.id,
+                    comment.author,
+                    comment.message,
+                    str(comment.created_at)))
+
+            return result
+
+        except sqlalchemy.exc.SQLAlchemyError as alchemy_ex:
+            msg = str(alchemy_ex)
+            LOG.error(msg)
+            raise shared.ttypes.RequestFailed(shared.ttypes.ErrorCode.DATABASE,
+                                              msg)
+
+        except Exception as ex:
+            msg = str(ex)
+            LOG.error(msg)
+            raise shared.ttypes.RequestFailed(shared.ttypes.ErrorCode.IOERROR,
+                                              msg)
+
+    @timeit
+    def getCommentCount(self, bug_hash):
+        """
+            Return the number of comments for the given bug.
+        """
+        session = self.__session
+        try:
+            commentCount = session.query(Comment) \
+                .filter(Comment.bug_hash == bug_hash) \
+                .count()
+
+            if commentCount is None:
+                commentCount = 0
+
+            return commentCount
+        except sqlalchemy.exc.SQLAlchemyError as alchemy_ex:
+            msg = str(alchemy_ex)
+            LOG.error(msg)
+            raise shared.ttypes.RequestFailed(shared.ttypes.ErrorCode.DATABASE,
+                                              msg)
+
+        except Exception as ex:
+            msg = str(ex)
+            LOG.error(msg)
+            raise shared.ttypes.RequestFailed(shared.ttypes.ErrorCode.IOERROR,
+                                              msg)
+
+    @timeit
+    def addComment(self, bug_hash, comment_data):
+        """
+            Add new comment for the given bug.
+        """
+        session = self.__session
+        comment = Comment(bug_hash,
+                          comment_data.author,
```

Comment:
Security leak! 

Author must not be given by the client! It must be queried from the authenticated session. Wire in comment_data.author="Anonymous" for now and add a FIXME and a followup issue in GITHUB to query the proper user name.